docs_src/design/adr/core/0021-Device-Profile-Changes.md

```diff
@@ -0,0 +1,64 @@
+# Changes to Device Profiles
+
+## Status
+** Proposed **
+(as of 1/20/22)
+
+Please see a [prior PR](https://github.com/edgexfoundry/edgex-docs/pull/605) on this topic that detailed much of the debate and context on this issue.  For clarity and simplicity, that PR was closed in favor of this simpler ADR.
+
+## Context
+While the device profile has always been the way to describe a device/sensor and template its communications to the rest of the EdgeX platform, over the course of EdgeX evolution there have been changes in what could change in a profile (often based on its associations to other EdgeX objects).  This document is meant to address the issue of change surrounding device profiles in EdgeX going forward – specifically when can a device profile (or its sub-elements such as device resources) be added, modified or removed.
+
+### Summary of Device Profile Rules
+These rules will be implemented in core metadata on device profile API calls.
+
+- A device profile can be added anytime
+- Device resources or device commands can be added to a device profile anytime
+- Attributes and properties can be added to a device profile anytime
+- A device profile can be modified when the device profile is not associated to a device or provision watcher
+    - this includes modifying any properties (except identifiers like names and ids)
+    - this includes changes to the array of device resources, device commands
+    - this includes changes to attributes and properties (of device resources)
+    - even when a device profile is associated to a device or provision watcher, properties of the device profile or device resource can be modified when the property change will not affect the behavior of the system (example: the description property has no effect on the system).
+- A device profile cannot be removed when it is associated to a device or provision watcher.
+- A device profile can be removed or modified even when associated to an event or reading unless configuration options (see New Configuration Settings below) are set prohibiting removal or modification when associated to an event/reading
```

Comment:
I thought the `New Configuration Settings` was just to allow/disallow in general (beyond the other rules) since we don't know what  `event/reading` may have been exported that the adopter cares about. i.e is is not just about what is stored in Core Data.

Reply:
@lenny-intel - I changed this a bit to reflect your rationale.  The fact is that with the new config, it simply disallows the change/removal (period, end of line).  There was a rationale for this config which was to preserve the association to events/readings, but this just goes to explain why it is there.  It doesn't change anything if a profile is associated to an event / reading or not.